shared_test.go

```diff
@@ -21,6 +21,7 @@ type ComplexRecord struct {
 	Count     int       `csv:"count"`
 	Rate      float32   `csv:"rate"`
 	Tags      string    `csv:"tags"`
+	Rank      uint      `csv:"rank"`
```

Comment:
Please use a uint8 so you could also use 1000 as an invalid value, and simulate an issue an int64 might have with a huge number